QMigrate/io/event.py

```diff
@@ -387,19 +384,57 @@ def write(self, run):
 
         out = {"EventID": self.uid, **self.trigger_info, **self.localmag}
         out = {**out, **self.max_coalescence}
-        for _, location in self.locations.items():
-            out = {**out, **location}
+
+        # Rename keys for locations; do not output covariance loc (just err)
+        loc = self.locations["spline"]
+        gau = dict((f"GAU_{key}", value) \
+            for (key, value) in self.locations["gaussian"].items())
+        cov = dict((f"COVARIANCE_{key}", value) \
+            for (key, value) in list(self.locations["covariance"].items())[3:])
+        out = {**out, **loc, **gau, **cov}
 
         if self.localmag.get("ML") is not None:
             event_file_cols = EVENT_FILE_COLS + ["ML", "ML_Err", "ML_r2"]
         else:
             event_file_cols = EVENT_FILE_COLS
 
-        eventfile_df = pd.DataFrame([out])[event_file_cols]
+        event_df = pd.DataFrame([out])[event_file_cols]
+
+        # Set floating point precision for COA values
+        for col in event_df.filter(like="COA").columns:
+            event_df[col] = event_df[col].map(lambda x: f"{x:.4g}",
+                                              na_action="ignores")
+
+        # Set floating point precision for locations & loc uncertainties
+        ll_corner = lut.coord2grid(lut.ll_corner, inverse=True)
+        for i, axis in enumerate(["X", "Y", "Z"]):
+            # Calculate appropriate number of decimal places based on LUT cell
+            # size and coordinate projection
+            index = np.zeros(3, dtype=int)
+            index[i] = 1
+            diff = (ll_corner - lut.index2coord(index))[0][i]
+            dec_pos = int(np.format_float_scientific(diff).split("e")[1])
+            # Sort out which columns to format
+            cols = [axis, f"GAU_{axis}"]
+            if axis == "Z":
+                decimals = max((dec_pos * -1 + 2), 0)
+                cols.extend(event_df.filter(regex="Err[X,Y,Z]"))
+            else:
+                decimals = max((dec_pos * -1 + 2), 6)
+            for col in cols:
+                event_df[col] = event_df.loc[:, col].round(decimals=decimals)
+                if decimals <= 0:
+                    event_df[col] = event_df.loc[:, col].astype(int)
+
```

Comment:
Flagging this as a reminder to add a property to the LUT class that returns `dec_pos`, something like

```python
dec_pos = lut.precision
...
   class LUT:
        ...
        @property
        def precision(self):
            """
            Get appropriate number of decimal places as a function of the
            cell size and coordinate projection.
            """

            ll_corner = self.coord2grid(self.ll_corner, inverse=True)
            one_in = lut.index2coord([1, 1, 1])
            return [int(np.format_float_scientific(axis).split("e")[1])
                    for axis in (ll_corner - one_in)[0]]
```

Reply:
This looks good. The current implementation will be retained for this PR, but can be replaced with this more efficient method after merging changes from #75

Note: for precision, the value should be multiplied by -1:

```
            return [int(np.format_float_scientific(axis).split("e")[1]) for axis
                    in np.subtract(*self.index2coord([[0, 0, 0], [1, 1, 1]]))] * -1
```

Reply:
Good catch, added to my branch